### DISCUSSION TITLE
Tycho 4.0.0 release schedule?

### DISCUSSION THREAD

**Discussion by tja-m** [eclipse-tycho/tycho#2387] [General]
Is there a schedule, when Tycho 4.0.0 is planned to be release?

I recall reading something like "should be part of March release", but couldn't find it anymore.

I am very much interested in using Tycho's new support for "CI friendly versions", but couldn't find anything on a release plan.

**Comment by laeubi** [comment]
There is no real plan, but likely will be somewhere after the 2023-06 release if no big points are needed until there.

You can always use the current [tycho snapshot](https://github.com/eclipse/tycho/wiki#getting-tycho-snapshots) build, this helps us to make sure no "surprise" happens after the release, just keep in mind that for ci friendly versions/extensions in general you need to configure the repo in the settings.xml:  https://github.com/eclipse-tycho/tycho/wiki#configure-snapshots-repository-globally

Beside that there is also a version of ci friendly versions in the 3.x releases.

**Comment by tja-m** [comment]
Thank you for your quick response!

In that case I will make a try building my code branch with the "CI friendly" adaptions with Tycho 3.0.4 first.

(Builds are already successful with Tycho 4.0.0-SNAPSHOT, at least locally)

**Comment by tja-m** [comment]
OK I tried with 3.0.4 and as far as I have my Jenkins pipeline yet, everything seems to have worked (injecting version via `-Drevision=` ...).

So here is my contribution on testing 4.0.0-SNAPSHOT:

I set up two branches now, which differ in the Tycho version only (3.0.4 vs. 4.0.0-SNAPSHOT). 
Only with 4.0.0-SNAPSHOT there is a problem in resolving the target platform:

```
[INFO] --- tycho-compiler:4.0.0-SNAPSHOT:validate-classpath (default-validate-classpath) @ com.somecompany.someapp.common.util ---
[INFO] Resolving class path of com.somecompany.someapp.common.util
[INFO] Downloading from p2: https://<my-targetplatform-location>/plugins/ch.qos.logback.slf4j_1.2.3.jar
[ERROR] An error occurred while transferring artifact canonical: osgi.bundle,ch.qos.logback.slf4j,1.2.3 from repository https://<my-targetplatform-location>:
[ERROR]    download from https://<my-targetplatform-location>/plugins/ch.qos.logback.slf4j_1.2.3.jar failed

...

[ERROR] Failed to execute goal org.eclipse.tycho:tycho-compiler-plugin:4.0.0-SNAPSHOT:validate-classpath (default-validate-classpath) on project com.somecompany.someapp.common.util: Execution default-validate-classpath of goal org.eclipse.tycho:tycho-compiler-plugin:4.0.0-SNAPSHOT:validate-classpath failed: Could not mirror artifact osgi.bundle,ch.qos.logback.slf4j,1.2.3 into the local Maven repository.See log output for details.: No credentials provided -> [Help 1]
```

The credentials are provided through environment variables and propagated in the global settings.xml:

```xml
        <server>
            <id>com.somecompany.someapp.server.artifactory.p2-targetplatform</id>
            <username>${env.ARTIFACTORY_CREDENTIALS_USR}</username>
            <password>${env.ARTIFACTORY_CREDENTIALS_PSW}</password>
        </server>
```

The target platform definition contains:

```xml
<?xml version="1.0" encoding="UTF-8" standalone="no"?>
<?pde version="3.8"?><target name="Some App Target" sequenceNumber="77">
    <locations>
        <location includeAllPlatforms="false" includeConfigurePhase="true" includeMode="planner" includeSource="true" type="InstallableUnit">

            <repository
            	id="com.somecompany.someapp.server.artifactory.p2-targetplatform"
                location="https://<my-targetplatform-location>"/>

            <unit id="com.somecompany.someapp.target.feature.feature.group" version="1.2.3"/>
            <unit id="com.somecompany.someapp.target.feature.source.feature.group" version="1.2.3"/>
        </location>
    </locations>
</target>
```

The build is executed in Jenkins Pipeline (with Docker agent) using a Docker image derived from 
`maven:${mavenVersion}-eclipse-temurin-${jdkVersion}-focal` running

* Current temurin jdk 17
* Maven 3.9.1

As stated above the same constellation works with Tycho 3.0.4.
